### PR TITLE
fix(code): Updated Michigan opinion scrapers

### PR DESCRIPTION
**Pull request by satsuki-chan**
Update to scrape opinions from the new websites for the **Michigan Supreme and Appellate** courts.

- **Supreme** court:
https://www.courts.michigan.gov/case-search/?resource=supreme-court-opinion&page=1&resultType=opinions&sortOrder=Newest&pageSize=100&aAppellateCourt=Supreme%20Court
- **Appellate** court:
https://www.courts.michigan.gov/case-search/?resource=opinion&page=1&resultType=opinions&sortOrder=Newest&pageSize=100&aAppellateCourt=Court%20Of%20Appeals

Both scrapers for the Appellate court were consolidated in one.

```
juriscraper.opinions.united_states.state.mich
juriscraper.opinions.united_states.state.michctapp
```

Fixes: #487

### CODE REVIEW

File: juriscraper/opinions/united_states/state/mich.py [hunk 1]
```diff
@@ -3,132 +3,75 @@
...
-                end_year=self.today.year,
-            )
-        )
-        self.back_scrape_iterable = list(range(0, 868))
         self.court_id = self.__module__
+        self.url = "https://www.courts.michigan.gov/api/CaseSearch/SearchCaseOpinions?resource=supreme-court-opinion&aAppellateCourt=Supreme%20Court&searchQuery=&sortOrder=Newest&page=1&pageSize=100"
+        self.back_scrape_iterable = list(range(1, 99))
+        self.request["headers"] = {"User-Agent": ""}
+        self.title_re = (
+            r"(MSC|COA) (?P<docket>\d+)\s+(?P<name>.+)\s+(Opinion|Order)"
```
Comment (flooie) [inline juriscraper/opinions/united_states/state/mich.py:27]: Let's not merge court of appeals into the Supreme Court scraper.

File: juriscraper/opinions/united_states/state/michctapp.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                    "name": name,
+                    "url": url,
+                    "lower_courts": lower_courts,
+                    "status": status,
+                }
+            )
+
+    def get_status(self, title: str) -> str:
+        if " Published" in title:
+            status = "Published"
```
Comment (flooie) [inline juriscraper/opinions/united_states/state/michctapp.py:57]: can you update the doc strings 

And on a side note is the white space in the If statement needed.

File: juriscraper/opinions/united_states/state/mich.py (hunk 1 above)
Comment (satsuki-chan) [reply juriscraper/opinions/united_states/state/mich.py]: Regular expression for both courts separated

File: juriscraper/opinions/united_states/state/michctapp.py (hunk 2 above)
Comment (satsuki-chan) [reply juriscraper/opinions/united_states/state/michctapp.py]: Added examples of expected titles to check the document status.

Removed the space before the check for the "Published" status.

**Pull request merged**